### ISSUE TITLE
box_download_options - allow for the '~' to shortcut to home dir

### ISSUE THREAD

**Issue by esend7881** [hashicorp/vagrant#11819]
Please note that the Vagrant issue tracker is in priority reserved for bug reports and enhancements. For general usage questions, please use
HashiCorp Discuss: https://discuss.hashicorp.com/c/vagrant/24 Thank you!

When submitting a bug report, please provide the minimal configuration and required information necessary to reliably reproduce the issue. It
should include a basic Vagrantfile that only contains settings to reproduce the described behavior.

**Tip:** Before submitting your issue, don't hesitate to remove the above introductory text, possible empty sections (e.g. References), and this tip.

### Vagrant version

2.2.9

### Host operating system

Windows 10

### Vagrantfile

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "RMBN/0.0.1"

  config.vm.provider :virtualbox do |virtualbox, override|
    override.vm.box_download_options = {
      "netrc-file": "#{ENV['userprofile']}\\.netrc"
    }
    override.vm.box_url = "https://path/to/my/box/rmbn-box-0.0.1.box"
  end
end
```

### Debug output

This works, the problem is that on Windows I need to write in `#{ENV['userprofile']}\\.netrc`

### Expected behavior

What I think should happen is the value `~/.netrc` should work whether on linux or Windows. I.e., if my host is a Windows machine and `~` is picked up on, it should replace it with the `%userprofile%` variable. 

I.e., it would be much cleaner if all you needed was:
```
    override.vm.box_download_options = {
      "netrc-file": "~/.netrc"
    }
```

I have a feeling the code where this kind of parsing can be done is here:
https://github.com/hashicorp/vagrant/blob/8951975d779fa54482fa22958109a9ecc5904044/lib/vagrant/util/downloader.rb#L249

So in short, this is more like a feature request to ensure that, in Windows, if `~` is part of a path name, it should be replaced by `%USERPROFILE%`

**Comment by chrisroberts** [comment]
Hi there,

Unfortunately, these options are not something we would want to modify as this option hash is used to pass custom flags to the download executable. The key/value pairs are simply formatted for passing to the CLI, there is no builtin knowledge of what the options provided are, or if there is any reason we should be modifying them.

However, you can update your Vagrantfile so that the path expansion is consistent:

```ruby
override.vm.box_download_options = {
  "netrc-file" => File.expand_path("~/.netrc")
}
```

Cheers!

**Comment by esend7881** [comment]
> Hi there,
> 
> Unfortunately, these options are not something we would want to modify as this option hash is used to pass custom flags to the download executable. The key/value pairs are simply formatted for passing to the CLI, there is no builtin knowledge of what the options provided are, or if there is any reason we should be modifying them.
> 
> However, you can update your Vagrantfile so that the path expansion is consistent:
> 
> ```ruby
> override.vm.box_download_options = {
>   "netrc-file" => File.expand_path("~/.netrc")
> }
> ```
> 
> Cheers!

Very cool! Did not know about the `File` api
